src/utils/api.js

```diff
@@ -0,0 +1,121 @@
+import { processServerResponse } from "./promise";
+import { API_KEY } from "./constants";
+
+/*const baseUrl =
+  process.env.REACT_APP_API_URL === "production"
+    ? "https://api.reciperoulette.twilightparadox.com"
+    : "http://localhost:3002"; */
+
+const baseUrl = import.meta.env.VITE_APP_API_URL;
+
+const searchRecipes = async (searchTerm, page) => {
+  try {
+    const url = new URL(`${baseUrl}/api/recipes/search`);
+    url.searchParams.append("searchTerm", searchTerm);
+    url.searchParams.append("page", page.toString());
+
+    const response = await fetch(url.toString());
+
+    if (!response.ok) {
+      throw new Error(`HTTP error. Status: ${response.status}`);
+    }
+
+    const text = await response.text();
```

Comment:
Please, use `processServerResponse` everywhere not to duplicate the `ok` checking